### PR TITLE
adding changes to update a PR

### LINKED ISSUE #59: If PR exists, update title and body
In my workflow, the PR title and body are dynamically generated based on the commits contained within. I would love it if this action could update the title and body of the PR the PR is already found.

### PR DESCRIPTION
**Pull request by vsoch**
This PR will:
 - add an ability to specify a PR should be updated to close #59, `PULL_REQUEST_UPDATE` set to anything indicates an update is desired, but only if the PR exists.
 - including a second envar to set a state, `PULL_REQUEST_STATE`, to update the state only if an update is done 
 - refactor of main function into smaller functions
 - adding docstrings

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

**Comment by shaungrady** [comment]
- Getting an error when it tries to update the PR: https://github.com/shaungrady/tmp/runs/1864725515?check_suite_focus=true#step:4:37

- The action also fails if a PR cannot be created if the branches are even: https://github.com/shaungrady/tmp/runs/1864707419?check_suite_focus=true#step:4:31

  > No commits between main and test

  Shouldn't fail the action, I think.

**Comment by vsoch** [comment]
The error on update looks like I can fix - when they are equal / even, what outcome would you expect?

**Comment by shaungrady** [comment]
Maybe just a log message?

**Comment by vsoch** [comment]
That sounds good! I'll see if I can work on this after the workday today. Thanks for testing it out!

**Comment by vsoch** [comment]
@shaungrady for the second, I think maybe the appropriate thing to do would be to set `PASS_ON_ERROR` ? It could be that someone else would want it to fail given this condition.

**Comment by vsoch** [comment]
was building a container and had a minute :) You should be able to test again and set `PASS_ON_ERROR` for the second case.

**Comment by shaungrady** [comment]
I don't think I'd want it to pass on error. But the "no commits between branches" error doesn't feel like an error—it's just a skip condition.

**Comment by vsoch** [comment]
It's technically a 412 response and the message is included in the "errors" list, so I think I'd consider it an error. This PR here is to just address doing an update, so if you want to open another issue and propose a plan for custom parsing of errors (and get others support) we can address it there.

**Comment by shaungrady** [comment]
It succeeds, then throws a 422 `Fork collab can only be enabled on cross-repo pull requests` error.

https://github.com/shaungrady/tmp/runs/1868039744#step:4:32

**Comment by vsoch** [comment]
Oh interesting, I wonder if it’s an issue with setting maintainer can modify, since it’s not relevant for a PR to your own repository.

**Comment by vsoch** [comment]
Ok should be ready to try again. If we get the same issue I’ll open a support request with GitHub.

**Comment by shaungrady** [comment]
Error's gone, but it still appears to be exiting with a non-zero code: https://github.com/shaungrady/tmp/runs/1868623416#step:4:1

File: pull-request.py [hunk 1]
```diff
@@ -49,13 +70,196 @@ def parse_into_list(values):
...
+        "title": title,
+        "body": body,
+        "base": target,
+        "state": state or "open",
+    }
+    # PATCH /repos/{owner}/{repo}/pulls/{pull_number}
+    url = "%s/%s" % (PULLS_URL, entry.get("number"))
+    print("Data for updating pull request: %s" % data)
+    response = requests.patch(url, json=data, headers=HEADERS)
+    if response.status_code != 201:
```
Comment (vsoch) [inline pull-request.py:138]:
```suggestion
    if response.status_code != 200:
```

**Comment by vsoch** [comment]
That should be easy to fix! We were expecting 201 (akin to the creation endpoint) but looks like it should be 200. I just made this change:

https://github.com/vsoch/pull-request-action/pull/60/commits/a1e87fb17a32a11827f2773eb9192253b5fa8827

**Comment by shaungrady** [comment]
Looks like it's all fixed now—good to go 👍

**Comment by vsoch** [comment]
okay, let's give it a shot then!

**Pull request merged**